component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/OneTimePassword.java

```diff
@@ -0,0 +1,248 @@
+            num /= 10;
+            if (doubleDigit) {
+                digit = doubleDigits[digit];
+            }
+            total += digit;
+            doubleDigit = !doubleDigit;
+        }
+        int result = total % 10;
+        if (result > 0) {
+            result = 10 - result;
+        }
+        return result;
+    }
+
+    /**
+     * This method uses the JCE to provide the HMAC-SHA-1
+     * algorithm. HMAC computes a Hashed Message Authentication Code and in this
+     * case SHA1 is the hash algorithm used.
+     *
+     * @param keyBytes the bytes to use for the HMAC-SHA-1 key
+     * @param text     the message or text to be authenticated.
+     * @throws NoSuchAlgorithmException if no provider makes either HmacSHA1 or HMAC-SHA-1 digest
+     *                                  algorithms available.
+     * @throws InvalidKeyException      The secret provided was not a valid HMAC-SHA-1 key.
+     */
+
+    public static byte[] hmacShaGenerate(byte[] keyBytes, byte[] text)
+            throws NoSuchAlgorithmException, InvalidKeyException {
+
+        Mac hmacSha;
+        try {
+            hmacSha = Mac.getInstance(VoiceOTPConstants.ALGORITHM_HMAC);
+        } catch (NoSuchAlgorithmException nsa) {
+            hmacSha = Mac.getInstance(VoiceOTPConstants.ALGORITHM_HMAC_SHA);
+        }
+        SecretKeySpec macKey = new SecretKeySpec(keyBytes, "RAW");
+        hmacSha.init(macKey);
+        return hmacSha.doFinal(text);
+    }
+
+    /**
+     * This method generates an OTP value for the given set of parameters.
+     *
+     * @param secret           the shared secret
+     * @param movingFactor     the counter, or other value that changes on a per use
+     *                         basis.
+     * @param codeDigits       the number of digits in the OTP, not including the checksum,
+     *                         if any.
+     * @param addChecksum      a flag that indicates if a checksum digit
+     *                         should be appended to the OTP.
+     * @param truncationOffset the offset into the MAC result to begin truncation. If this
+     *                         value is out of the range of 0 ... 15, then dynamic truncation
+     *                         will be used. Dynamic truncation is when the last 4 bits of
+     *                         the last byte of the MAC are used to determine the start
+     *                         offset.
+     * @throws NoSuchAlgorithmException if no provider makes either HmacSHA1 or HMAC-SHA-1 digest
+     *                                  algorithms available.
+     * @throws InvalidKeyException      The secret provided was not a valid HMAC-SHA-1 key.
+     */
+    public static String generateOTP(byte[] secret, long movingFactor, int codeDigits, boolean addChecksum,
+                                     int truncationOffset) throws NoSuchAlgorithmException, InvalidKeyException {
+        // put movingFactor value into text byte array
+        String result = null;
```

Comment:
```suggestion
    public static String generateOTP(byte[] secret, long movingFactor, int codeDigits, boolean addChecksum,
                                     int truncationOffset) throws NoSuchAlgorithmException, InvalidKeyException {
                                     
        // Put movingFactor value into text byte array.
        String result = null;
```

Check for similar formatting errors in the PR